### ISSUE TITLE
Add support for access points in EFS file system

### ISSUE THREAD

**Issue by mungoknotwise** [terraform-providers/terraform-provider-aws#12118]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Amazon Elastic File System (Amazon EFS) Access Points is a new EFS feature that simplifies providing applications access to shared data sets in an EFS file system. EFS Access Points work together with AWS IAM and enforce an operating system user and group, and a directory for every file system request made through the access point.  

EFS Access Points represent a new way to manage application access in NFS environments with increased scalability, security, and ease of use. Use cases that can benefit from EFS Access Points include container-based environments where developers build and deploy their own containers, data science applications that require access to production data, and sharing a specific directory in your file system with other AWS accounts.

### New or Affected Resource(s)

* resource: aws_efs_access_point
* data source: aws_efs_access_point
* data source: aws_efs_access_point_ids

### Potential Terraform Configuration

```hcl
resource "aws_efs_access_point" "this" {
  file_system_id = "sample"
  client_token   = "sample token"
  posix_uid      = 1001
  posix_gid      = 1001
  secondary_gids = [1003, 1004]
  path           = "/sample"
  owner_uid      = 1001
  owner_gud      = 1001
  permissions    = 755
  tags = {
    Name = "example"
  }
}

output "apid" {
  value = aws.efs_access_point.access_point_id
}

data "aws_efs_access_point" "by_apid" {
  access_point_id = "fsap-12345"
}

data "aws_efs_access_point_ids" "by_fsid" {
  file_system_id  = "sample"
}

output "apids" {
  value = data.aws_efs_access_point_ids.by_fsid.ids
}
```

### References

* https://aws.amazon.com/blogs/aws/new-for-amazon-efs-iam-authorization-and-access-points/

**Comment by bflad** [comment]
The maintainers would prefer the plural data source to be named `aws_efs_access_points` over `aws_efs_access_point_ids` as it may return information beyond just IDs.

**Comment by bflad** [comment]
Hi folks 👋  The `aws_efs_access_point` resource and singular data source were previously released in version 2.63.0 of the Terraform AWS Provider. Thanks to @DrFaust92 for the implementation there. Support for the plural `aws_efs_access_points` data source has been merged and will release with version 2.68.0 of the Terraform AWS Provider, likely later today. 👍
